spec/models/customer_spec.rb

```diff
@@ -69,4 +69,45 @@
       expect(c.user).to eq user2
     end
   end
+
+  describe 'scopes' do
+    context 'managed_by' do
+      let!(:admin) { create(:admin_user) }
+      let!(:user) { create(:user) }
+      let!(:enterprise) { create(:enterprise, owner: user) }
+      let!(:customer) { create(:customer, enterprise: enterprise, user: user) }
+      let!(:customer1) { create(:customer, enterprise: enterprise) }
+
+      let!(:user1) { create(:user) }
+      let!(:enterprise1) { create(:enterprise, owner: user1) }
+      let!(:customer2) { create(:customer, enterprise: enterprise1, user: user1) }
+
+      # manager of enterprise1
+      let!(:user2) { create(:user) }
+
+      # user who has edit profile permission on enterprise1
+      let!(:user3) { create(:user) }
+      let!(:enterprise2) { create(:enterprise, owner: user3) }
```

Comment:
I could be wrong, but I don't think these need to be `let!`. If changed to just `let`, they will only execute when needed and save time.

Reply:
Also is the comment about edit permission right? It is `user2` that has edit_profile permission, not `user3`.

Reply:
Thanks for clarifying, but I think some of the variables aren't used in all tests, so I've added a commit to explain what I mean.